092-self-healing-using-cruise-control.md

```diff
@@ -0,0 +1,253 @@
+In the same way user can enable and disable the self-healing for other anomalies.
+
+### Using Notifiers when using Self-healing
+
+Users can set the notifier as per their choice.
+They can also use the notifier provided by Cruise Control like the `SelfHealingNotifier`, `SlackSelfHealingNotifier`, `AlertaSelfHealingNotifier`, `MSTeamsSelfHealingNotifier` or they can even build own custom notifier jar and use it.
+In case, the user wants to have some Kubernetes way to see what anomalies are being detected or the self-healing operations are being performed on the cluster, they can use our custom `KubernetesEventsNotifier` which will be available under the Strimzi GitHub organization.
+
+#### KubernetesEventsNotifier 
+
+Cruise Control provides `AnomalyNotifier` interface which has multiple abstract methods on what to do if certain anomalies are detected.
+`SelfHealingNotifier` is the base class that contains the logic of self-healing and override the methods in by implementing the `AnomalyNotifier` interface.
+Some of those methods are:`onGoalViolation()`, `onBrokerFailure()`, `onDiskFailure`, `alert()` etc.
+The `KubernetesEventsNotifier` is based of the `SelfHealingNotifier` class.
+For effective alerts and monitoring, it makes use of Kubernetes events to signal to users what operations Cruise Control is performing automatically.
+The `KubernetesEventsNotifier` will override the `alert()` method of the `SelfHealingNotifier` and will trigger and publish events whenever an anomaly is detected.
+These events would publish information regarding the anomaly like the anomalyID, the type of anomaly which was detected etc.
+The events would help the user to understand the anomalies that are present, when the anomalies appeared in their cluster and also would help them know what is going on in their clusters.
+
+The event published would look like this:
+```yaml
+- action: <action>
+  apiVersion: v1
+  eventTime: "2024-12-19T08:06:21.314140Z"
+  firstTimestamp: null
+  involvedObject:
+    kind: Pod
+    name: <cruise-control-pod-name>
+    namespace: myproject
+  kind: Event
+  lastTimestamp: null
+  message: Cruise Control is fixing the detected anomaly
+  metadata:
+    creationTimestamp: "2024-12-19T08:06:21Z"
+    generateName: cruise-control-event
+    name: my-cluster-34c41626-0add-47a3-9b79-d6ee29d5d81f  # my-cluster-<Anomaly ID>
+    namespace: myproject
+    resourceVersion: "101070"
+    uid: 80892c7d-e3a1-44ad-84dc-2d88e15a7ff0
+  reason: Anomaly was detected in the cluster GOAL_VIOLATION # Detected Anomaly type
+  reportingComponent: cruise-control
+  reportingInstance: cruise-control
+  source: {}
+  type: Normal
+```
+
+The users can build a jar for the notifier for inclusion in the Kafka image to run on as part of Cruise Control.
```

Comment:
The `KubernetesEventsNotifier` should be included out of the box within the Kafka Strimzi image in order to be configured by the operator. It's not clear if you are referring to a custom notifier that a user would write.